google/cloud/bigtable/examples/bigtable_samples_instance_admin_ext.cc

```diff
@@ -0,0 +1,305 @@
+            return i.name() == instance_name;
+          });
+  // [END bigtable_check_instance_exists]
+
+  // Create instance if does not exists
+  if (not instanceExists) {
+    // [START bigtable_create_prod_instance]
+    std::cout << "\nCreating a PRODUCTION Instance: ";
+    google::cloud::bigtable::DisplayName display_name("Sample Instance");
+
+    // production instance needs at least 3 nodes
+    auto cluster_config = google::cloud::bigtable::ClusterConfig(
+        zone, 3, google::cloud::bigtable::ClusterConfig::SSD);
+    google::cloud::bigtable::InstanceConfig config(
+        google::cloud::bigtable::InstanceId(instance_id), display_name,
+        {{cluster_id.get(), cluster_config}});
+    config.set_type(google::cloud::bigtable::InstanceConfig::PRODUCTION);
+
+    auto instance_details = instance_admin.CreateInstance(config).get();
+    std::cout << " Done" << std::endl;
+    // [END bigtable_create_prod_instance]
+  } else {
+    std::cout << "\nInstance " << instance_id.get() << " already exists."
+              << std::endl;
+    return;
+  }
+
+  // [START bigtable_list_instances]
+  std::cout << "\nListing Instances: " << std::endl;
+  auto instances_after = instance_admin.ListInstances();
+  for (auto const& instance : instances_after) {
+    std::cout << instance.name() << std::endl;
+  }
+  // [END bigtable_list_instances]
+
+  // [START bigtable_get_instance]
+  std::cout << "\nGet Instance: " << std::endl;
+  auto instance = instance_admin.GetInstance(instance_id.get());
+  std::string instance_detail;
+  google::protobuf::TextFormat::PrintToString(instance, &instance_detail);
+  std::cout << "GetInstance details : " << std::endl << instance_detail;
+  // [END bigtable_get_instance]
+
+  // [START bigtable_get_clusters]
+  std::cout << "\nListing Clusters: " << std::endl;
+  auto cluster_list = instance_admin.ListClusters(instance_id.get());
+  std::cout << "Cluster Name List: " << std::endl;
+  for (auto const& cluster : cluster_list) {
+    std::cout << "Cluster Name: " << cluster.name() << std::endl;
+  }
+  // [END bigtable_get_clusters]
+}
+
+void CreateDevInstance(std::string project_id, int argc, char* argv[]) {
+  if (argc != 4) {
+    throw Usage{
+        "create-dev-instance: <project-id> <instance-id> <cluster-id> <zone>"};
+  }
+  google::cloud::bigtable::InstanceId instance_id(ConsumeArg(argc, argv));
+  google::cloud::bigtable::ClusterId cluster_id(ConsumeArg(argc, argv));
+  std::string const zone = ConsumeArg(argc, argv);
+
+  // Create an instance admin endpoint.
+  google::cloud::bigtable::InstanceAdmin instance_admin(
+      google::cloud::bigtable::CreateDefaultInstanceAdminClient(
+          project_id, google::cloud::bigtable::ClientOptions()));
+
+  std::cout << "\nCheck Instance exists: " << std::endl;
+  auto instances = instance_admin.ListInstances();
+  auto instance_name =
+      instance_admin.project_name() + "/instances/" + instance_id.get();
+  bool instanceExists =
```

Comment:
Please follow the C++ naming conventions here.

Reply:
fixed.